packages/web/src/components/session-menu/secrets-panel.tsx

```diff
@@ -0,0 +1,139 @@
+// import { SessionMenuPanelContentsProps } from '.';
+
+import { useCallback, useEffect, useState } from "react";
+import { toast } from 'sonner';
+import { type SecretWithAssociatedSessions } from '@srcbook/shared';
+
+import { getSecrets, associateSecretWithSession, disassociateSecretWithSession } from "@/lib/server";
+import { Switch } from "@/components/ui/switch";
+import { SessionMenuPanelContentsProps } from ".";
+
+type PropsType = Pick<SessionMenuPanelContentsProps, 'session'>;
+
+export default function SessionMenuPanelSecrets(props: PropsType) {
+  const [secretsList, setSecretsList] = useState<
+    | { status: 'idle' } 
+    | { status: 'loading' }
+    | { status: 'complete', data: Array<SecretWithAssociatedSessions> }
+    | { status: 'error' }
+  >({ status: 'idle' });
+  useEffect(() => {
+    const run = async () => {
+      setSecretsList({ status: 'loading' });
+      let result: Array<SecretWithAssociatedSessions>;
+      try {
+        result = (await getSecrets()).result;
+      } catch (err) {
+        console.error('Error loading secrets!', err);
+        setSecretsList({ status: 'error' });
+        return;
+      }
+
+      setSecretsList({ status: 'complete', data: result });
+    };
+
+    // FIXME: it is possible for this to run twice in parallel?
+    run();
+  }, []);
```

Comment:
I couldn't find an example of data loading happening in anywhere in the application outside ofa react-router `loader` function. Given that this secrets list data is really only used in the secrets list panel, it seemed like the logical place to manage it, but in order to kick off the data fetching from here I had to add some relatively complex state management.

I can do a few things potentially here:
1. Leave it as it is (fwiw, I tend to be a fan of what I did here for its explicitness, but I recognize I'm probably in the minority on this view 😄 )
2. Switch this over to some other pre-built componet level data fetching tool like swr or react-query
3. Move this into the session page `loader` function and drill the state all the way down to this component, plus a way to "expire" it or oprimisitcally update it when changes are made to the switches

Reply:
No this is great for now. We have held off on fancier data fetching / state management solutions until we:

1. Have lots of examples in our code base to best inform us how we'd like to handle this
2. Have any sort of perf considerations (right now, we only run locally, so caching or anything clever isn't that helpful)

:shipit: